### ISSUE TITLE
Unexpected behaviour with Observable.take(1).repeat(2) cannot be subscribed to more than once.

### ISSUE THREAD

**Issue by RoryKelly** [ReactiveX/RxPY#187]
x = Observable.from_(["hi"]).take(1).repeat(2)

x.subscribe(on_next=print, on_error=print)  # emits hi hi
x.subscribe(on_next=print, on_error=print)  # just completes
x.subscribe(on_next=print, on_error=print)  # just completes

**Comment by thomasnield** [comment]
I'll take a look today

**Comment by thomasnield** [comment]
Yes, the `repeat()` operator is broken for multiple Observers. A quick workaround you can use in the interim while myself or Dag tries to fix this is to wrap everything in an `Observable.defer()`. This will create a fresh new `Observable` operation each time. 

```python
x = Observable.defer(lambda: Observable.from_(["hi"]).take(1).repeat(2))
```

**Comment by thomasnield** [comment]
I fixed this issue and put in PR #189 

See changes here:
https://github.com/ReactiveX/RxPY/pull/189/commits/cd84e2118163add3d746cd2a483e249a4bace8ae

Essentially, I wrapped the implementation for `repeat()` in an `Observable.defer()` to ensure a new operation instance is created for each `Observer`.
